include/tatami/isometric/binary/arith_helpers.hpp

```diff
@@ -27,7 +27,9 @@ struct DelayedBinaryArithHelper {
     /**
      * @cond
      */
-    static constexpr bool always_sparse = (op_ != DelayedArithOp::DIVIDE);
+    static constexpr bool always_sparse = (op_ == DelayedArithOp::ADD ||
+                                           op_ == DelayedArithOp::SUBTRACT ||
+                                           op_ == DelayedArithOp::MULTIPLY);
```

Comment:
Seems like `POWER` would be okay here. The rule of thumb is whether `OP(0, 0) == 0`.

Reply:
(Same could be said for `delayed_arith_always_sparse`, as long as `POWER` and `right_ = true` without special values.